packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -291,6 +302,9 @@ export class QueuedRequestController extends BaseController<
       this.#processingRequestCount += 1;
       try {
         await requestNext();
+        if (request.method === "wallet_switchEthereumChain") {
+          this.#flushQueueForOrigin(request.origin);
```

Comment:
The selected network can change in other circumstances as well. For example if the user switches the network dropdown in the popup, according tot he designs it should also update the dapp-selected network (I know that's not implemented yet, but it will be soon).

So we may want to trigger this flushing somewhere else, which would capture both cases. e.g. when the state changes, or in a method responsible for making that state change

Reply:
hmm yeah good thinking ahead there. Might be worth experimenting to see if we can put this on a 'selectedNetwork changed' event handler instead.